### ISSUE TITLE
PSP is not correctly applied to istiod pod in Istio

### ISSUE THREAD

**Issue by mjakobczyk** [kyma-project/kyma#10894]
**Description**

In Kyma there are two main predefined PodSecurityPolicy's: unprivileged (**001-kyma-unprivileged**) and privileged (**002-kyma-privileged**). We expect that each deployed pod will be granted with one of these. By default pods should not have privileged access to disallow making any trouble.

**Expected result**

An expected situation is that pods that belong to Istio (**istio-ingressgateway** and **istiod**) are assigned to **001-kyma-unprivileged** PSP. **istio-egressgateway** is not considered as we do not validate output traffic.

**Actual result**

Unfortunately, while deploying Kyma ootb **istiod** is not always assigned to the correct PSP.

**Steps to reproduce**

1. Provision GKE cluster from Gardener
2. Deploy Kyma from master on freshly-created cluster
3. Check annotations of **istiod** pod
`kubectl get pod -n istio-system <istiod-pod-name> -o json | jq '.metadata.annotations'`
4. Check **istiod** securityContext from container specs
`kubectl get pod -n <istiod-pod-name> -o json | jq '.spec.containers[0].securityContext'`

It looks like on Gardener **istiod** pod is assigned to Gardener's PSP (which is provided ootb and we can not manage it anyhow). After comparing securityContext it looks like it's kind of similar to the one that we use, but not the same.

**Troubleshooting**

After deeper investigation it looks like the issue managing configuration of the cluster itself that enforces unprivileged access can solve the issue. In order to test it go to Gardener's YAML configuration and set: `allowPrivilegedContainers: false`.

**Comment by strekm** [comment]
for both ingress-gateway and istio we have applied the same securityContext configuration:
```
  securityContext:
    runAsUser: 65534
    runAsNonRoot: true
    runAsGroup: 65534
```
for some reason ingress-gateway gets `privileged:false` and istiod is not getting that property. From Istio point of view we cannot set explicitly `privileged:false` cause istio operator configuration doesn't expose it. 
https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#PodSecurityContext

**Comment by aberezovski** [comment]
@strekm @mjakobczyk 
Hi guys,
I've executed the command mentioned in the issue description over the Kyma 1.22.0 release candidate cluster, but was not able to see the annotations with validated PSPs.
```
kubectl get pod -n istio-system istiod-5d9f946cd7-9gjld  -o json | jq '.metadata.annotations'
{
  "sidecar.istio.io/inject": "false"
}
```
I was able to see an incomplete securityContext for the istiod running pod, but that appears to be an expected behavior.
```
kubectl get pod -n istio-system istiod-5d9f946cd7-9gjld  -o json | jq '.spec.containers[0].securityContext'  
{
  "capabilities": {
    "drop": [
      "ALL"
    ]
  },
  "runAsGroup": 1337,
  "runAsNonRoot": true,
  "runAsUser": 1337
}
```

On another hand, I could not see the validated PSP details in any pod' annotations running in `kube-system` on Kyma 1.22.0 release cluster.
Did I do something wrong, or release cluster was provisioned in a way that does not support that?

**Comment by strekm** [comment]
@aberezovski you mean `kube-system` or `kyma-system`. are there any other pods which don't have PSP?

**Comment by mjakobczyk** [comment]
@aberezovski Could you check if there is any PSP present in the cluster? `kubectl get psp --all-namespaces`

**Comment by aberezovski** [comment]
@strekm , @mjakobczyk 
Hi guys,
It appears that Kyma 1.22.0 release candidate cluster has mis-configured PSPs. I've asked @adamwalach why that is happening.
On another hand I've provisioned a new gardener cluster and installed Kyma 1.22.0 on top of it and was able to see the proper PSP related info in the pods' annotations.
I also saw that `istiod` pods was validated by the `gardener.privileged` PSP that is quite permissive in comparison with our intention to reduce the privilege containers.

I asked Istio community about [What is the most restricted PSP for Istio Control Plane istiod pod](https://discuss.istio.io/t/what-is-the-most-restricted-psp-for-istio-control-plane-istiod-pod/10401). Let's see  if we get an official confirmation about what is the accepted approach.
